### DISCUSSION TITLE
SRD table footnotes

### DISCUSSION THREAD

**Discussion by Craftidore** [Obsidian-TTRPG-Community/TTRPG-Community-Admin#21] [The Burned Steak]
*Obsidian's clipboard-html-to-markdown converter is a library called [Turndown](https://help.obsidian.md/Obsidian/Credits#Turndown). I'll be referring to it as Turndown throughout this and don't want anyone to be confused.*

## Situation

In the [PF1E SRD vault](https://github.com/Obsidian-TTRPG-Community/Pathfinder-1E-SRD-Markdown), I'm currently working on Rules → Magic Items. The magic item rules very frequently have tables with footnotes, like the following: 

![2023-05-25-213813_659x174_scrot](https://github.com/Obsidian-TTRPG-Community/TTRPG-Community-Admin/assets/73705352/da042b79-42cd-4786-9264-0b0908b76c21)

![2023-05-25-213933_702x206_scrot](https://github.com/Obsidian-TTRPG-Community/TTRPG-Community-Admin/assets/73705352/9c462471-8dfe-4c05-bba0-43afe17c9c56)

So, to recap how aonprd handles table footnotes:
- Single footnote: Use `*`
- More than 1 footnote: Use superscript numbers, starting at `1`

## Problems

This raises the question of how we should be handling these in the Obsidian SRD conversions. There are a couple problems here:

1. Not everyone uses Reading View
    - [Footnotes](https://help.obsidian.md/Editing+and+formatting/Basic+formatting+syntax#Footnotes) don't work in live-preview
2. Some of these tables are complicated enough that they have to be turned into html tables, and wiki links don't work in html tables
    - Thus ruling out [block-linking](https://help.obsidian.md/Linking+notes+and+files/Internal+links#Link+to+a+block+in+a+note) to each footnote

## My Current Solutions

Because I needed to work on SRD conversion, I went ahead with a 'solution' so I could keep working. 

### Single-Footnote tables:

Find the footnote in the table and replace it with `\*` (as Turndown doesn't catch this). Remove italics from footnote line (to match multi-footnote lines) and replace `*` with `\*`.

Result:

![2023-05-25-220551_1232x423_scrot](https://github.com/Obsidian-TTRPG-Community/TTRPG-Community-Admin/assets/73705352/03a91db4-93e8-4ade-8095-07cab64285cd)

### Multi-Footnote tables:

Find all footnotes in table (The regex \w\d\\| will catch them all, but will have false positives; should still make finding them easier. Regex is `\w\d|` in vim/neovim (but not Obsidian vim)), and surround each with `<sup></sup>`. 
Make all footnotes a unordered list item and add `:` after the number.

Result:

![2023-05-25-220806_1280x521_scrot](https://github.com/Obsidian-TTRPG-Community/TTRPG-Community-Admin/assets/73705352/cda04875-cd1e-4c02-8fd7-0559c79cc855)

(do excuse the messed-up html syntax highlighting; the markdown lsp I use has trouble with html sometimes)

### Other Notes on Table conversion

When converting a table, I paste within Obsidian and get a table like this:

```md
|   |   |
|---|---|
|**Header**|**Header**|
|Rest of table|Rest of Table|
```

I convert it to this:
```md
|**Header**|**Header**|
|---|---|
|Rest of table|Rest of Table|
```

- I leave the manually bolded table headers because I'm too lazy. 
- Any time there's italics in a table cell, Turndown decides to use `_` instead of `*`, which is normally fine, but with some tables, the `_` will break the table in Live Preview, so I've been find-and-replacing them with `*` so the table works in Live Preview.
- Also, table titles are sometimes H3s and sometimes just bold. I've been converting H3 table titles to normal bold text for consistency.

**Comment by sigrunixia** [comment]
You essentially did the same thing I would have done. I dont have any issues with this. What about everyone else?

**Comment by scottTomaszewski** [comment]
Yeah, I think this makes sense: its the most flexible while achieving the goal.  

I was debating suggesting consistency with the footnote "label" between single- and multi-footnote tables, but the more I think about it, asterisk for singles is effectively the same as a `1`. I was thinking about it in case content gets refactored in the future and then all the asterisks would need to get changed, but this content is pretty much set in stone at this point and really shouldn't be changing (that much).

Nice intuition, I like it :+1:
